Makefile.am

```diff
@@ -195,15 +195,21 @@ EXTRA_DIST += dist/initramfs-tools/scripts/init-premount/tpm2-totp
 
 if HAVE_MKINITCPIO
 helpers_SCRIPTS = dist/show-tpm2-totp
-initcpio_install_DATA = dist/initcpio/install/tpm2-totp
+initcpio_install_DATA = dist/initcpio/install/tpm2-totp dist/initcpio/install/sd-tpm2-totp
 initcpio_hooks_DATA = dist/initcpio/hooks/tpm2-totp
+
+systemdsystemunit_DATA = dist/tpm2-totp.service dist/tpm2-totp.timer
+install-data-hook:
+	mkdir -p $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
+	cd $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
+	$(LN_S) ../tpm2-totp.service && \
+	$(LN_S) ../tpm2-totp.timer
 if HAVE_PLYMOUTH
 initcpio_install_DATA += dist/initcpio/install/plymouth-tpm2-totp dist/initcpio/install/sd-plymouth-tpm2-totp
 initcpio_hooks_DATA += dist/initcpio/hooks/plymouth-tpm2-totp
 
-systemdsystemunit_DATA = dist/plymouth-tpm2-totp.service
+systemdsystemunit_DATA += dist/plymouth-tpm2-totp.service
 install-data-hook:
```

Comment:
The Makefile can only have one `install-data-hook`, so this fails if plymouth is installed because the first `install-data-hook` installing the symlinks for `tpm2-totp.service` and `tpm2-totp.timer` gets overwritten by this hook (which in turn fails because the `sysinit.target.wants` doesn't exist).

We need to split up the hook into separate targets that can be called from the single `install-data-hook`, e.g. like this:
```diff
diff --git a/Makefile.am b/Makefile.am
index a42e510..574f3a3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -199,7 +199,7 @@ initcpio_install_DATA = dist/initcpio/install/tpm2-totp dist/initcpio/install/sd
 initcpio_hooks_DATA = dist/initcpio/hooks/tpm2-totp
 
 systemdsystemunit_DATA = dist/tpm2-totp.service dist/tpm2-totp.timer
-install-data-hook:
+install-systemd-service-hook:
 	mkdir -p $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
 	cd $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
 	$(LN_S) ../tpm2-totp.service && \
@@ -209,9 +209,15 @@ initcpio_install_DATA += dist/initcpio/install/plymouth-tpm2-totp dist/initcpio/
 initcpio_hooks_DATA += dist/initcpio/hooks/plymouth-tpm2-totp
 
 systemdsystemunit_DATA += dist/plymouth-tpm2-totp.service
-install-data-hook:
+install-plymouth-service-hook:
+	mkdir -p $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
 	cd $(DESTDIR)$(systemdsystemunitdir)/sysinit.target.wants && \
 	$(LN_S) ../plymouth-tpm2-totp.service
+else
+install-plymouth-service-hook:
 endif # HAVE_PLYMOUTH
+
+install-data-hook: install-systemd-service-hook install-plymouth-service-hook
+
 endif # HAVE_MKINITCPIO
 EXTRA_DIST += dist/initcpio/hooks/tpm2-totp dist/initcpio/hooks/plymouth-tpm2-totp
```
(I opted to keep the `mkdir -p` in `install-plymouth-service-hook` since trying to create an already existing directory doesn't fail with `-p` and it feels cleaner to me to ensure that the necessary directory exists in every separate hook.)

Reply:
Sure, fixed it up. Thanks a lot!